### PR TITLE
Use recalculated_unit_cell in two_theta_refine

### PR DESCRIPTION
**Pull request by rjgildea**
Use `recalculated_unit_cell` (if set) in dials.scale and export_mtz.

Depends on cctbx/dxtbx#153

### CODE REVIEW

**Comment by rjgildea** [comment]
The test introduced in 0e9ef4b currently fails, as the `recalculated_unit_cell` is lost when `.change_basis` is called on the crystal in export_mtz:
https://github.com/dials/dials/blob/78f1e749952279caff1e77ccd0035f1b78f9cea8/util/export_mtz.py#L531
Obviously we could quite easily apply the change of basis to the `recalculated_unit_cell` in the `Crystal.change_basis` method, although I'm not sure whether we can do anything sensible to the recalculated unit cell parameter/volume sd's? @dagewa any thoughts here? Should we just discard the sd's in `Crystal.change_basis`?

**Comment by dagewa** [comment]
I've not yet tested, but this looks like the right approach. At a low level, `dials.two_theta_refine` just updates the usual cell via the **B** matrix parameterisation. At the end we make the high level decision to copy the unit cells into the original experiments as recalculated cells.

**Comment by dagewa** [comment]
> Should we just discard the sd's in `Crystal.change_basis`?

I would say yes, because some change of basis operations might result in slightly different estimated σ values after refinement if that change had been made before refinement rather than after. I haven't actually tested that, but I would not be surprised if that's the case.

Being defensive, we should also wipe out the errors any time the recalculated cell setter is called.

**Comment by dagewa** [comment]
I think this fixes #670

File: algorithms/scaling/scaling_library.py [hunk 1]
```diff
@@ -381,7 +381,11 @@ def determine_best_unit_cell(experiments):
     experiments."""
     uc_params = [flex.double() for i in range(6)]
     for exp in experiments:
-        for i, p in enumerate(exp.crystal.get_unit_cell().parameters()):
+        if exp.crystal.get_recalculated_unit_cell() is not None:
+            unit_cell = exp.crystal.get_recalculated_unit_cell()
+        else:
+            unit_cell = exp.crystal.get_unit_cell()
```
Comment (Anthchirp) [inline algorithms/scaling/scaling_library.py:387]: maybe
```suggestion
        unit_cell = exp.crystal.get_recalculated_unit_cell() or exp.crystal.get_unit_cell()
```
but you iirc you don't like that syntax.

File: util/export_mtz.py [hunk 2]
```diff
@@ -558,9 +558,12 @@ def export_mtz(integrated_data, experiment_list, params):
         else:
             experiment.data["ROT"] = z
 
-    mtz_writer.add_crystal(
-        params.mtz.crystal_name, unit_cell=experiment_list[0].crystal.get_unit_cell()
-    )  # Note: add unit cell here as may have changed basis since creating mtz.
+    if experiment_list[0].crystal.get_recalculated_unit_cell() is not None:
+        unit_cell = experiment_list[0].crystal.get_recalculated_unit_cell()
+    else:
+        unit_cell = experiment_list[0].crystal.get_unit_cell()
```
Comment (Anthchirp) [inline util/export_mtz.py:564]: same

File: algorithms/scaling/scaling_library.py (hunk 1 above)
Comment (rjgildea) [reply algorithms/scaling/scaling_library.py]: I think in this case it is probably justified as it is a simple enough assignment and makes the code much shorter.

**Pull request merged**